### ISSUE TITLE
Inconsistent responses to /connect API

### ISSUE THREAD

**Issue by Ambros94** [ACINQ/eclair#1030]
### Description
Responses to the API '/connect' are not reliable, most time it returns :
`
400 BAD REQUEST
{
    "error": "The form field 'uri' was malformed:\nSubstream Source cannot be materialized more than once"
}`

All the performed request are equal.

### Expected behaviour

* 200 OK "connected" at the first time it is called
* 200 OK "already connected" all the following times

### Actual behaviour

The first call of the API once the node is started is consistent: it returns "connected".
The following calls response are inconsistent, some undetected internal behaviour makes them vary between:
* 200 OK "already connected" all the following times
* 400 BAD REQUEST `{    "error": "The form field 'uri' was malformed:\nSubstream Source cannot be materialized more than once"}`

### How to reproduce

The issue can be reproduced simulating multiple calls to the connect API. I attach the postman and the CURL version I used during my tests. 
(The basic token matches )

curl -u :changeit -X POST -F uri=02acb5b3bdf46d3a35e665c9abe46def12c341f55d603853ddeb1371076cab5c57@68.183.221.146:49735 localhost:8080/connect

(My apologies to the guy running this node)

### Eclair Version and environment 
Eclair 2.11 Self-compiled from Github Master branch using: 
* Scala 2.12.8
* Java 11, openjdk 11.0.2 2019-01-15
* MacOs 10.14.5

### Extra information:

It looks like that if the call is not performed for more than 1 minutes, the following call will return the correct message "already connected".

Eclair.log does not contain any line regarding this connection and nothing is printed once an API call is performed.

**eclair.conf**:

`eclair.bitcoind.rpcuser="foo"
eclair.bitcoind.rpcpassword="bar"
eclair.bitcoind.rpcport="34897"
eclair.bitcoind.host="127.0.0.1"
eclair.chain="testnet"
eclair.api.enabled=true
eclair.api.password="changeit"`

**Comment by araspitzu** [comment]
Hi, I'm having some difficulty reproducing your issue, can you share how you make multiple calls to the API? Are those parallel or sequential? I have tried simulating several calls very close to each other and i always get the correct response, either:
- ``` {"error":"there is another connection attempt in progress"}``` 
- ```already connected```

The problem you encounter seem to stem from how we unmarshall request parameters (see [here](https://github.com/akka/akka-http/issues/745)), i've made a quickfix for that and if you're able to reproduce the issue consistently you can try running branch [akka_strict_entity_reading](https://github.com/ACINQ/eclair/tree/akka_strict_entity_reading)

**Comment by Ambros94** [comment]
TL;DR: It fixes the problem.

Thanks for the fast answer. 
It seems that the problem is related to the client performing the request. I was able to reproduce the issue using both postman and curl. The time it happens are way more frequently using curl (curl 7.54.0 (x86_64-apple-darwin18.0) libcurl/7.54.0 LibreSSL/2.6.5 zlib/1.2.11 nghttp2/1.24.1).

Using the following python code it works both on master and akka_strict_entity_reading:
```
import requests

for i in range(0,100):
    url = "http://localhost:8080/connect"

    payload = "------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name=\"uri\"\r\n\r\n02acb5b3bdf46d3a35e665c9abe46def12c341f55d603853ddeb1371076cab5c57@68.183.221.146:49735\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW--"
    headers = {
        'content-type': "multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW",
        'Authorization': "Basic OmNoYW5nZWl0",
        'cache-control': "no-cache"
        }

    response = requests.request("POST", url, data=payload, headers=headers)

    print(response.text)
```

Using the following bash script **it works only on akka_strict_entity_reading**:
```
for i in {1..100}; do curl -u :changeit -X POST -F uri=02acb5b3bdf46d3a35e665c9abe46def12c341f55d603853ddeb1371076cab5c57@68.183.221.146:49735 localhost:8080/connect; done
```

**Comment by Ambros94** [comment]
I am reopening the issues since I was able to reproduce it in completely a completely different environment, which I am about to describe:

### Eclair Version and environment

- eclair-node-0.3-a5debcd.jar (Downloaded from github release page)
- OpenJDK 11.0.3 2019-04-16
- Ubuntu 18.04.2 LTS
- bitcoind 0.18

- Vultr Cloud compute ($10/mo)
- 1vCPU
- 2048MB Memory
- 55GB SSD

Using both the scripts provided in the [comment above](https://github.com/ACINQ/eclair/issues/1030#issuecomment-500893957) I was able to reproduce the issue.

**Comment by Ambros94** [comment]
The issue does not appear using:

openjdk 11.0.3 2019-04-16
Ubuntu 19.04
eclair-node-0.3-a5debcd.jar (Downloaded from github release page)
bitcoind 0.18
RAM 4 GB
Processor: 1 vCores (2.4 GHz)
Storage:
20 GB SSD
Bandwidth:
100 Mbps
